rebalance/rebalance.py

```diff
@@ -168,16 +174,43 @@ def rebalance(plugin, outgoing_scid, incoming_scid, msatoshi: Millisatoshi = Non
     description = "%s to %s" % (outgoing_scid, incoming_scid)
     invoice = plugin.rpc.invoice(msatoshi, label, description, retry_for + 60)
     payment_hash = invoice['payment_hash']
-    success_msg = ""
-    try:
-        excludes = []
-        # excude all own channels to prevent unwanted shortcuts [out,mid,in]
-        mychannels = plugin.rpc.listchannels(source=my_node_id)['channels']
-        for channel in mychannels:
-            excludes += [channel['short_channel_id'] + '/0', channel['short_channel_id'] + '/1']
 
+    rpc_result = None
+    count = 0
+    excludes = [my_node_id]  # excude all own channels to prevent shortcuts
+    nodes = {}               # here we store erring node counts
+    routes = {}              # trace all routes to suppress duplicates
+    maxhops = 1              # start with short routes and increase
+    msat_factor = 4          # start with high msatoshi factor to reuce
+                             # WIRE_TEMPORARY failures because of imbalances
+
+    try:
         while int(time.time()) - start_ts < retry_for and not plugin.rebalance_stop:
-            r = plugin.rpc.getroute(incoming_node_id, msatoshi, riskfactor=1, cltv=9, fromid=outgoing_node_id, exclude=excludes)
+            count += 1
+            try:
+                r = plugin.rpc.getroute(incoming_node_id,
+                                        fromid=outgoing_node_id,
+                                        exclude=excludes,
+                                        msatoshi=msatoshi * msat_factor,
+                                        maxhops=maxhops,
+                                        riskfactor=10, cltv=9)
+            except RpcError as e:
+                # could not find route -> change params and restart loop
+                if e.method == "getroute" and e.error.get('code') == 205:
+                    # reducce msatoshi factor to look for smaller channels now
+                    msat_factor -= 1
+                    if msat_factor == 1:
+                        # when we reached neutral msat factor:
+                        # increase maxhops and restart with high msatoshi factor
+                        maxhops = maxhops + 1
+                        msat_factor = 4
+                        continue
```

Comment:
The condition should be `if msat_factor == 0:`.  Otherwise we will never check the neutral msat factor 1.

Reply:
good catch!

About the default values: sure it can be configurable and turned of per default.
But: we should test something like this in practice before hiding it behind a config switch that nobody uses ;) 

In the mailing list there's some chat about it that the (only?) potential drawback is computational time used for getroute.
If that's true then its really a matter of CPU vs Net-Latency tradeoff which in a lot situations will be no problem for the CPU.